### ISSUE TITLE
Proposal to Add Lip Vertex Error (LVE)  Metrics for 3D Talking Heads Evaluation

### ISSUE THREAD

**Issue by rittik9** [Lightning-AI/torchmetrics#3003]
## 🚀 Feature

I would like to propose incorporating an essential evaluation metric for 3D talking heads into the TorchMetrics library: Lip Vertex Error (LVE) . This metric is widely used in speech-driven facial animation research to assess lip synchronization accuracy respectively.

### Motivation

Current TorchMetrics offerings lack dedicated metrics for evaluating 3D talking heads, particularly in assessing the quality of lip synchronization . I think this metric also fits in multimodal folder of this library.

### Pitch

The Lip Vertex Error (LVE) metric evaluates the quality of lip synchronization in 3D facial animations by measuring the maximum Euclidean distance (L2 error) between corresponding lip vertices of the generated and ground truth meshes for each frame. This metric assesses how accurately the animated lip movements align with the intended speech input.

### References

Paper: [MeshTalk](https://openaccess.thecvf.com/content/ICCV2021/papers/Richard_MeshTalk_3D_Face_Animation_From_Speech_Using_Cross-Modality_Disentanglement_ICCV_2021_paper.pdf)

### Additional context

I would like to open a PR for the same.

**Comment by SkafteNicki** [comment]
@rittik9 sound good to me, feel free to open a PR !

**Comment by hugleecool** [comment]
when ,very need,

**Comment by rittik9** [comment]
Hi @hugleecool I'm occupied with some other stuff at this moment, So I'm  planning this for mid April. But if you are willing to open a pr, please go ahead, I'll be happy to help :)

**Comment by hugleecool** [comment]
sorry, I'm very poor in code, a product manager, learning ongoing

Rittik Panda ***@***.***> 于2025年4月5日周六 11:14写道：

**Comment by SkafteNicki** [comment]
@rittik9 I am wrong for thinking the metric is simply calculated as:
```python
def lip_vertex_error(vertices_pred: torch.Tensor, vertices_gt: torch.Tensor, mouth_map: list[int]) -> torch.Tensor:
    """
    Computes Lip Vertex Error (LVE) as used in CodeTalker.

    Args:
        vertices_pred (Tensor): (T, V, 3) predicted vertices
        vertices_gt (Tensor): (T, V, 3) ground truth vertices
        mouth_map (list[int]): list of mouth vertex indices

    Returns:
        Tensor: scalar LVE value
    """
    # Difference only for mouth vertices
    diff = vertices_gt[:, mouth_map, :] - vertices_pred[:, mouth_map, :]  # (T, M, 3)
    sq_dist = torch.sum(diff ** 2, dim=-1)  # (T, M)
    max_per_frame = torch.max(sq_dist, dim=1).values  # (T,)
    return torch.mean(max_per_frame)
```
if so I can probably do it fairly easy within the next couple of days

**Comment by rittik9** [comment]
@SkafteNicki Sure pls....
You can use this as a [reference](https://github.com/Doubiiu/CodeTalker/blob/main/main/cal_metric.py)
